### PR TITLE
BugFix/FixIsAvailableTimeout is available timeout

### PR DESCRIPTION
**Pull request by xieshouyu**
## What is the purpose of the change
When the zookeeper is disconnected, the dubbo actor check registry health will be stuck instead of returning to Down
I hope the result is to return to the Down state

The reason is that the isAvailable method calls the getServices method of zookeeper, causing continuous retries. When determining whether it is available, it is necessary to first determine whether it is connected before obtaining it.

## Brief changelog

1、Fix the issue of timeout for all calls to the isAvailable method after the zookeeper is disconnected
2、Move the isAvailable of NopServiceDiscovery into its own implementation class

## Verifying this change
Start the service, use pfctl or iptables to disconnect zookeeper, and it can return to the down state normally.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/ServiceDiscovery.java [hunk 1]
```diff
@@ -94,6 +95,13 @@ default long getDelay() {
         return getUrl().getParameter(REGISTRY_DELAY_NOTIFICATION_KEY, 5000);
     }
 
+    /**
+     * Get services is the default way for service discovery to be available
+     */
+    default boolean isAvailable() {
+        return CollectionUtils.isNotEmpty(getServices());
```
[the remark this answers is not in the document]
Comment (wxbty) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/ServiceDiscovery.java:102]: This is a new method, it seems that it just replaces the empty judgment of getServices().

**Review by wxbty — APPROVED** [review]

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/ServiceDiscovery.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlbumenJ) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/ServiceDiscovery.java:102]: But from the definition of `isAvailable`, it is necessary to improve the `available` definition

[the remark this answers is not in the document]
Comment (xieshouyu) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/ServiceDiscovery.java:102]: > check if destoryed here

👌🏻

**Review by AlbumenJ — APPROVED** [review]
LGTM. But pls merge into 3.2

**Pull request closed without merging**